File: src/main/java/io/lsdconsulting/lsd/distributed/interceptor/interceptor/EventPublisherInterceptor.java
package io.lsdconsulting.lsd.distributed.interceptor.interceptor;

import io.lsdconsulting.lsd.distributed.interceptor.captor.messaging.MessagingCaptor;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.messaging.Message;
import org.springframework.messaging.MessageChannel;
import org.springframework.messaging.support.ChannelInterceptor;

@Slf4j
@RequiredArgsConstructor
public class EventPublisherInterceptor implements ChannelInterceptor {

    private final MessagingCaptor captor;

    @Override
    public Message<?> preSend(Message<?> message, MessageChannel channel) {
        captor.capturePublishInteraction(message);
        return message;
    }
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/interceptor/interceptor/EventPublisherInterceptor.kt
package io.lsdconsulting.lsd.distributed.interceptor.interceptor

import io.lsdconsulting.lsd.distributed.interceptor.captor.messaging.MessagingCaptor
import org.springframework.messaging.Message
import org.springframework.messaging.MessageChannel
import org.springframework.messaging.support.ChannelInterceptor

class EventPublisherInterceptor(private val captor: MessagingCaptor) : ChannelInterceptor {
    override fun preSend(message: Message<*>, channel: MessageChannel): Message<*> {
        captor.capturePublishInteraction(message)
        return message
    }
}
